python/ec2_no_internet_access-triggered.py

```diff
@@ -0,0 +1,185 @@
+      return self.evaluate_route_table(configurationItem)
+      
+    # check the subnet for potential internet accessibility 
+    if configurationItem['resourceType'] == "AWS::EC2::Subnet":
+      # check if subnet has configured public ip assignment as default
+      if configurationItem['configuration']['mapPublicIpOnLaunch']:
+        return 'NON_COMPLIANT'
+        
+      # check if subnet has a route to an internet gateway
+      try: 
+        route_table = self.get_related_configuration_item(self.find_relationships_by_type('AWS::EC2::RouteTable').pop())
+      except:
+        # no routing table associated, get main routing table of VPC
+        vpc = self.get_related_configuration_item(self.find_relationships_by_type('AWS::EC2::VPC').pop())
+        route_tables = self.find_relationships_by_type('AWS::EC2::RouteTable', vpc['relationships'])
+        for i in route_tables:
+          r = self.get_related_configuration_item(i)
+          if r['configuration']['associations'][0]['main']:
+            route_table = r
+            break
+        else:
+          raise Exception('Main route table not found', vpc)
+         
+      # check if assigned route table has a rule with an internet gateway
+      return self.evaluate_route_table(route_table)
+    
+    # check if the instance has a public ip assigned
+    if configurationItem['resourceType'] == 'AWS::EC2::Instance':
+      if configurationItem['configuration']['publicIpAddress']:
+        return 'NON_COMPLIANT'
+      return 'COMPLIANT'
+      
+    # check if network interface has a public ip associated
+    if configurationItem['resourceType'] == 'AWS::EC2::NetworkInterface':
+      for i in configurationItem['configuration']['privateIpAddresses']:
+        if i['association']:
+          return 'NON_COMPLIANT'
+      return 'COMPLIANT'
+      
+    return 'NOT_APPLICABLE'
+  
+  def evaluate_route_table(self, route_table):
+    for route in route_table['configuration']['routes']:
+      if route['gatewayId'] and route['gatewayId'].startswith('igw-'):
+        return 'NON_COMPLIANT'
+    return 'COMPLIANT'  
+
+
+def lambda_handler(event, context):
+  try:
+    invokingEvent = json.loads(event['invokingEvent'])
+    configurationItem = invokingEvent['configurationItem']
+  except:
+    raise Exception('Could not load configuration item', event)
+ 
+  try:
+    rule = RaiseInternetConnectivity(configurationItem)
+  except:
+    raise Exception('Could not process configuration item', configurationItem)
+  
+  compliance = rule.evaluate_compliance()
+ 
+  print('Compliance evaluation for %s: %s' % (configurationItem['resourceId'], compliance))
+ 
+  # inform config rules about our evaluation result
+  rule.put_evaluations(compliance, event['resultToken'])
+
+  event_detail = configurationItem
+  event_detail['evaluationResult'] = compliance
+
+  # publish a message with the evaluation result to SNS if a topic has been configured
+  if SNS_TOPIC != 'INSERT_SNS_TOPIC_ARN':
```

Comment:
Most of the rule looks good. I wanted to point out that this is actually unnecessary. If you specify an SNS Topic in the AWS Config settings, Config Rules (both Custom and Managed) will send their evaluation results to that SNS topic each time PutEvaluations is called.